operations/security_and_authentication/Add_LDAP_User_Federation.md

```diff
@@ -1,32 +1,90 @@
 
+
 ### Prerequisites
 
 LDAP user federation is not currently configured in Keycloak. For example, if it was not configured in Keycloak when the system was initially installed or the LDAP user federation was removed.
 
-### Procedure
-
-1.  Update the LDAP settings in the customizations.yaml file.
-
-    The LDAP server CA certificate goes into SealedSecret. Refer to the "Generate Sealed Secrets" header in the [Prepare Site Init](../../install/prepare_site_init.md) procedure for instructions on how to do that.
-
-    1.  Update the customizations.yaml file.
 
-        LDAP connection information is stored in the keycloak-users-localize Secret in the services namespace. In the customizations.yaml file, set the values for the keycloak\_users\_localize keys in the spec.kubernetes.sealed\_secrets field:
-
-        -   The ldap\_connection\_url key is required and is set to an LDAP URL.
-        -   The ldap\_bind\_dn and ldap\_bind\_credentials keys are optional.
-        -   If the LDAP server allows anonymous searches of users and groups, then these keys must not be set.
-        -   If the LDAP server requires authentication. then the bind DN and credentials are set in these keys respectively.
-        For example:
+### Procedure
 
-        ```bash
-        keycloak_users_localize:
+1. Prepare to customize the customizations.yaml file.
+
+   If the customizations.yaml file is managed in an external Git repository (as recommended), then clone a local working tree. Replace the `<URL>` value in the following command before running it.
+
+   ```bash
+   ncn-m001# git clone <URL> /root/site-init
+   ncn-m001# cd /root/site-init
+   ```
+
+   If there is not a backup of site-init, perform the following steps to create a new one using the values stored in the Kubernetes cluster.
+
+   1. Create a new site-init directory using from the CSM tarball.
+
+      ```bash
+      ncn-m001# cp -r ${CSM_DISTDIR}/shasta-cfg /root/site-init
+      ncn-m001# cd /root/site-init
+      ```
+  
+   1. Extract customizations.yaml from the site-init secret.
+
+      ```bash
+      ncn-m001# kubectl -n loftsman get secret site-init -o jsonpath='{.data.customizations\.yaml}' | base64 -d - > customizations.yaml
+      ```
+
+   1. Extract the certificate and key used to create the sealed secrets.
+
+      ```bash
+      ncn-m001# mkdir certs
+      ncn-m001# kubectl -n kube-system get secret sealed-secrets-key -o jsonpath='{.data.tls\.crt}' | base64 -d - > certs/sealed_secrets.crt
+      ncn-m001# kubectl -n kube-system get secret sealed-secrets-key -o jsonpath='{.data.tls\.key}' | base64 -d - > certs/sealed_secrets.key
+      ```
+
+  > **NOTE:** All subsequent steps of this procedure should be performed within the `/root/site-init` directory created in this step.
+   
+2. Update the LDAP settings in the customizations.yaml file.
+
+   1. Add the encrypted version of the LDAP CA certificate in the `certs.jks` section of customizations.yaml.
```

Comment:
Does this need to explain how to generate the certs.jks here?
Also, this isn't encrypted at this point, it's base-64 encoded.

The procedure for generating certs.jks are in https://github.com/Cray-HPE/docs-csm/blob/main/install/prepare_site_init.md , in the part starting "If LDAP requires TLS (recommended), update the cray-keycloak sealed secret value by supplying a base64 encoded Java KeyStore (JKS) that contains the CA certificate that signed the LDAP server's host key."

Also, this should say that this is optional just like the "Prepare Site Init" section.

Reply:
If it is an optional step, then I think it might make more sense to just reference the procedure in the install file.  We can mark it as optional and say something like the following instead of listing out the actual commands:

> 2.1. (Optional) Add the LDAP CA certificate in the `certs.jks` section of customizations.yaml.
> Follow step 8 under the "Create Baseline System Customizations" header in the [Prepare Site Init](../../install/prepare_site_init.md) procedure.

Does this seem like a reasonable approach?